FilePersistence/src/version_control/GitRepository.cpp

```diff
@@ -318,30 +318,48 @@ CommitGraph GitRepository::commitGraph(QString startRevision, QString endRevisio
 
 const Commit* GitRepository::getCommit(QString revision) const
 {
-	Q_ASSERT(revision != WORKDIR);
 	Q_ASSERT(revision != INDEX);
+	if (revision == WORKDIR)
+	{
+		Commit* m = nullptr;
```

Comment:
??? This should be `Commit* m = new Commit{}` or even better `auto result = new Commit{}`. Either way it should be initialized. Right now you're writing somewhere in zero memory. Did you test this code?